### PR TITLE
Add support for Asia Pacific (Osaka) region

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Closes https://github.com/weaveworks/eksctl/issues/3031.

The `ap-northeast-3` region is not opt-in, so it can use the standard account ID, which means adding support for it is very simple (a little too simple, let me know if there is more to it 😅 ).

Fargate support in this region will lag behind for a couple of days, so there is now a check on create to log this. I have also added a warning in the docs, but since this will be removed in like a week, I wonder if it is necessary? Maybe simply pinning an issue in this repo will be enough to cover us for that time 🤷‍♀️ .

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/eks/fargate.go [hunk 1]
```diff
@@ -64,12 +65,15 @@ func DoCreateFargateProfiles(config *api.ClusterConfig, fargateClient FargateCli
...
-		switch errors.Cause(err).(type) {
-		case nil:
+		switch {
+		case err == nil:
 			logger.Info("created Fargate profile %q on EKS cluster %q", profile.Name, clusterName)
-		case *eks.ResourceInUseException:
+		case errors.As(err, &e):
 			logger.Info("Either Fargate profile %q already exists on EKS cluster %q or another profile is being created/deleted, no action taken", profile.Name, clusterName)
+		case fargate.IsUnauthorizedError(err):
+			return errors.Wrapf(err, "either account is not authorized to use Fargate or region %s is not supported", config.Metadata.Region)
```
Comment (aclevername) [inline pkg/eks/fargate.go:76]: big nit: can we add a unit test :smile:

**Review by aclevername — APPROVED** [review]
LGTM

File: pkg/eks/fargate.go (hunk 1 above)
Comment (Callisto13) [reply pkg/eks/fargate.go:76]: yep. we do have a test which check for failing errors (unlike an other in this switch which would not fail only log), so it is kindof covered, but i can add something explicit for this error

Comment (Callisto13) [reply pkg/eks/fargate.go:76]: the only difference here is that we alter the error message

**Pull request merged**